chapter4.adoc

```diff
@@ -221,9 +220,54 @@ follows:
 ], config:{lanes: 1, hspace:1024}}
 ....
 
+=== MTT access permissions lookup process
+
+MTT access permissions for a physical address PA in the context of a
+supervisor domain is ascertained as follows:
+
+1. Let a be `mttp.ppn`x`PAGESIZE`, and let i = LEVELS, where for mode
+Smmtt[34][*], LEVELS = 2 and for Smmtt[46, 56][*rw*] LEVELS = 3; PAGESIZE is
+2^12; MTT NON_LEAF_PTE_SIZE = 8 bytes. The `mttp` register must be active,
+i.e., the effective privilege mode must be HS-mode, S-mode or U-mode.
+
+2. Let mpte be the value of the MTT PTE at address a + pa.ppn[i] x
+NON_LEAF_PTE_SIZE. If accessing pte violates a PMA or PMP check, raise
+an access-fault exception corresponding to the original access type.
+
+3. If any bits or encodings that are reserved for future standard use are
+set within mpte, stop and raise a page-fault exception corresponding to
+the original access type.
+
+4. Otherwise, the mpte is valid. If (i=0) or (i=1 and mpte.type is not
+`MTT_L1_DIR`), go to step 5. Otherwise, the mpte is a pointer to the next
+level of the MTT. Let i = i - 1. If i < 0, stop and raise a page-fault
+exception corresponding to the original access type. Otherwise, let
+a = mpte.ppn x PAGESIZE and go to step 2. Note that when mpte.type =
+MTT_L1_DIR, the mpte.ppn field is the value of the mpte.info field.
+
+5. A leaf mpte has been found. Determine if the requested physical memory
+access is allowed per the access permission encoding, AND'ing the MTT
+permissions with the cumulative read, write, execute permissions derived from
```

Comment:
Why allow for write-only permissions? Why do the ANDing? If the read permission is not available then cause an access fault. If the access is an implicit or explicit write then cause an access fault.

Reply:
for implicit reads, writes that makes sense - for the final PA - the 1st stage and/or the G-stage permissions may disallow writes, but the MTT may allow rw - and the effective permission if any caching is performed should be the AND of the permissions accumulated from 1st, G and MTT.